public/pages/AnomalyCharts/utils/anomalyChartUtils.ts

```diff
@@ -138,3 +150,281 @@ export const disabledHistoryAnnotations = (
     },
   ];
 };
+
+export const ANOMALY_HEATMAP_COLORSCALE = [
+  [0, '#F2F2F2'],
+  [0.2, '#F2F2F2'],
+  [0.2, '#F7E0B8'],
+  [0.4, '#F7E0B8'],
+  [0.4, '#F2C596'],
+  [0.6, '#F2C596'],
+  [0.6, '#ECA976'],
+  [0.8, '#ECA976'],
+  [0.8, '#E78D5B'],
+  [1, '#E8664C'],
+];
+
+export enum AnomalyHeatmapSortType {
+  SEVERITY = 'by_severity',
+  OCCURRENCES = 'by_occurrences',
+}
+
+const getColorForValue = (value: number) => {
+  if (
+    value >=
+    ANOMALY_HEATMAP_COLORSCALE[ANOMALY_HEATMAP_COLORSCALE.length - 1][0]
+  ) {
+    return ANOMALY_HEATMAP_COLORSCALE[ANOMALY_HEATMAP_COLORSCALE.length - 1][1];
+  }
+
+  if (value <= ANOMALY_HEATMAP_COLORSCALE[0][0]) {
+    return ANOMALY_HEATMAP_COLORSCALE[0][1];
+  }
+
+  for (let i = 0; i < ANOMALY_HEATMAP_COLORSCALE.length - 1; i++) {
+    if (
+      value >= ANOMALY_HEATMAP_COLORSCALE[i][0] &&
+      value < ANOMALY_HEATMAP_COLORSCALE[i + 1][0]
+    ) {
+      return ANOMALY_HEATMAP_COLORSCALE[i + 1][1];
+    }
+  }
+};
```

Comment:
Could we just use a `switch` or cascading `if` statement here to determine the corresponding color? This would eliminate the need for `ANOMALY_HEATMAP_COLORSCALE` altogether, and make it easier to add a different color for any value > 0 to help solve the issue of low anomaly grades having the same color as anomaly grades of 0.

Reply:
I think using `switch` or cascading `if` may not be flexible when a different color/value is added. With current implementation, we don't need to make any change for this method if a new color/value is added. I will add comment to explain saying 1st `if ` is to check if value is larger than biggest value in color scale, 2nd `if` is to check value is smaller than smallest value. 

And the issue of low anomaly grades having 0 grade color is fixed by adding 2 lines above, to specify for number very very close to 0, give it color 0 grade color. This is some tricky way as specified [here](https://plotly.com/javascript/colorscales/#custom-discretized-heatmap-colorscale)
```
  [0.0000001, '#F2F2F2'],
  [0.0000001, '#F7E0B8'],

Reply:
I see, yeah that's fair enough. I think adding those 2 lines is a good solution